robottelo/utils/ohsnap.py

```diff
@@ -37,19 +37,39 @@ def ohsnap_repo_url(ohsnap_repo_host, request_type, product, release, os_release
             logger.warning(
                 'The snap version was not provided. Snap number will not be used in the URL.'
             )
-        release = release.split('.')
-        if len(release) == 2:
-            release.append('0')
-        release = '.'.join(release[:3])  # keep only major.minor.patch
+        if len(release.split('.')) == 2:
+            logger.warning(
+                f'.z version component not provided in the release ({release}),'
+                f' fetching the recent z-stream from ohsnap'
+            )
+            res, _ = wait_for(
+                lambda: requests.get(
+                    f'{ohsnap.host}/api/streams',
+                    hooks={'response': ohsnap_response_hook},
+                ),
+                handle_exception=True,
+                raise_original=True,
+                timeout=ohsnap.request_retry.timeout,
+                delay=ohsnap.request_retry.delay,
+            )
+            logger.debug(f'List of releases returned by Ohsnap: {res.json()}')
```

Comment:
I would move this section to a separate function in case we want to reuse `/api/streams` elsewhere in the future.

Reply:
This turned out to be a philosophical question rather than a technical one. Let's leave the action up to the next user who would like to use `/api/streams` API.